spec/santa_maria/unit/gateway/santa_maria_v2_spec.rb

```diff
@@ -286,4 +287,70 @@
       end
     end
   end
+
+  context 'color api' do
+    before do
+      response = {
+        colors: api_colors_response
+      }
+
+      stub_request(:get, "#{endpoint}api/v2/colors").with(
+        headers: {
+          'Accept-Language' => language,
+          'Channel' => 'flourishweb',
+          'Domaincode' => domaincode,
+          'X-Api-Key' => token
+        }
+      ).to_return(
+        body: response.to_json,
+        status: 200
+      )
+    end
+
+    context 'no colors' do
+      let(:api_colors_response) { [] }
+
+      it 'returns an empty array' do
+        expect(gateway.all_colors).to eq([])
+      end
+    end
+
+    context 'two colors' do
+      let(:api_colors_response) do
+        [
+          {
+            colorId: "1032530",
+            rgb: "B1AFB1",
+          },
+          {
+            colorId: "1032534",
+            rgb: "A1AD62"
+          }
+        ]
+      end
+
+      let(:expected_colors) do
+        [
+          {
+            color_id: "1032530",
+            rgb: "B1AFB1",
+          },
+          {
+            color_id: "1032534",
+            rgb: "A1AD62"
+          }
+        ]
+      end
+
+      it 'returns all colors' do
+        expected = expected_colors.map do |expected_color|
+          a_color(expected_color)
+        end
+
+        expect { |block| gateway.all_colors.each(&block) }.to (
+          yield_successive_args(*expected)
```

Comment:
This felt awful.  Can an RSpec expert provide advice on a better way to assert?  Wanted to continue along the custom matchers road.

Reply:
We've tended to just use descriptive methods (ie. `expect_colors_to_eq`) to assert rather than custom matchers, iirc @craigjbass had some experience with custom matchers but I'm not sure how much